### ISSUE TITLE
Popup does not fix horizontally when in Firefox overflow menu (OSX)

### ISSUE THREAD

**Issue by dotboris** [dotboris/prod-guard#437]
When you put Prod Guard in the Firefox overflow menu, the popup contents end up being too big. This leads to a horizontal scroll.

<details>
<summary>Screenshot</summary>

![osx popup scroll in overflow](https://user-images.githubusercontent.com/1874242/121810345-7cd7a000-cc2e-11eb-80e9-55550a7a351c.png)

</details>

This happens because the popup has a `min-width` of `25rem` which fits fine when on Linux but on OSX it does not. It seems that the width of the overflow menu on OSX is smaller than 25 rem.

**Comment by dotboris** [comment]
There's a deeper problem at hand here. An extension popup can be in one of two modes: In the overflow menu & not in the overflow menu.

When the popup is in the overflow menu, it is given a fixed width by Firefox. When it's not in the overflow menu, the size of the popup is dynamic. Firefox looks at the size of the content and sets the size of the popup based on that.

Depending on what "mode" you're in, you want to set the size of the popup differently. When in the overflow menu, you don't want to set a width and just let the content take as much horizontal space as possible. When outside of the overflow menu, you want to set the width so that the popup is not super small.

The current situation is that Prod Guard can't tell if its popup is in the overflow menu or not and therefore it can't tell if it should set a width or not.

From what I've been reading, figuring out if you're in the overflow menu or not is pretty difficult to pull off. I looked at the APIs web extensions have access to and it seems like there's nothing that will provide that information. I've also tried using something like `display: flex` to set a preferred size while still allowing to shrink & grow but it doesn't work really well.

There are two bugs in firefox that relate to this: 
- https://bugzilla.mozilla.org/show_bug.cgi?id=1373490
- https://bugzilla.mozilla.org/show_bug.cgi?id=1448276

I also looked at uBlock origin to see how they solved that problem. Turns out that when you open the popup, it runs a test to see if the content is bigger than the popup and deduces that it should go in a "portrait" mode when that happens. See: https://github.com/gorhill/uBlock/blob/3fa731606276217c8e022c1bdf418b4795359819/src/js/popup-fenix.js#L1324-L1356
